server/lib/availabilitySync.ts

```diff
@@ -57,141 +54,93 @@ class AvailabilitySync {
-            });
-
-            logger.info(
-              `Media ID ${media.id} does not exist in any of your media instances. Status will be changed to unknown.`,
-              { label: 'AvailabilitySync' }
-            );
-
-            await mediaRepository.update(media.id, {
-              status: MediaStatus.UNKNOWN,
-              status4k: MediaStatus.UNKNOWN,
-              serviceId: null,
-              serviceId4k: null,
-              externalServiceId: null,
-              externalServiceId4k: null,
-              externalServiceSlug: null,
-              externalServiceSlug4k: null,
-              ratingKey: null,
-              ratingKey4k: null,
-            });
-
-            await requestRepository.remove(request);
+            (media.status = mediaStatus),
+              (media.serviceId =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.serviceId
+                  : null),
+              (media.externalServiceId =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.externalServiceId
+                  : null),
+              (media.externalServiceSlug =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.externalServiceSlug
+                  : null),
+              (media.ratingKey =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.ratingKey
+                  : null);
           }
-        }
 
-        if (media.mediaType === 'tv') {
-          // ok, the show itself exists, but do all it's seasons?
-          const seasons = await seasonRepository.find({
-            where: [
-              { status: MediaStatus.AVAILABLE, media: { id: media.id } },
-              {
-                status: MediaStatus.PARTIALLY_AVAILABLE,
-                media: { id: media.id },
-              },
-              { status4k: MediaStatus.AVAILABLE, media: { id: media.id } },
-              {
-                status4k: MediaStatus.PARTIALLY_AVAILABLE,
-                media: { id: media.id },
-              },
-            ],
-          });
+          if (
+            media.status4k === MediaStatus.AVAILABLE ||
+            media.status4k === MediaStatus.PARTIALLY_AVAILABLE
+          ) {
+            (media.status4k = mediaStatus4k),
+              (media.serviceId4k =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.serviceId4k
+                  : null),
+              (media.externalServiceId4k =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.externalServiceId4k
+                  : null),
+              (media.externalServiceSlug4k =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.externalServiceSlug4k
+                  : null),
+              (media.ratingKey4k =
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.ratingKey4k
+                  : null);
+          }
```

Comment:
Switched to using query builder to make sure requests are related to only available media. We then check those requests to set the media to the correct status. For example, if the media is deleted and a request still remains, the media should be set to processing.